public/app/features/explore/state/query.ts

```diff
@@ -355,9 +340,34 @@ export const importQueries = (
  * @param modification Action object with a type, e.g., ADD_FILTER
  * @param modifier Function that executes the modification, typically `datasourceInstance.modifyQueries`.
  */
-export function modifyQueries(exploreId: ExploreId, modification: QueryFixAction, modifier: any): ThunkResult<void> {
-  return (dispatch) => {
-    dispatch(modifyQueriesAction({ exploreId, modification, modifier }));
+export function modifyQueries(
+  exploreId: ExploreId,
+  modification: QueryFixAction,
+  modifier: (query: DataQuery, modification: QueryFixAction) => Promise<DataQuery>
+): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    const state = getState().explore[exploreId];
+    if (state == null) {
+      return;
+    }
+
+    const { queries } = state;
+
+    const nextQueriesRaw = await Promise.all(queries.map((query) => modifier({ ...query }, modification)));
+
+    // we did some async calculations here,
+    // the redux-store could have changed in the meantime.
+    // we will check, and, if changed, we will not apply our changes.
+    const currentQueries = getState().explore[exploreId]?.queries ?? [];
```

Comment:
Was it added to cover some specific case? I wonder if this may lead to weird bugs where the list of queries doesn't actually change (just there's a reference to a new list)